qcodes/utils/dataset/doNd.py

```diff
@@ -483,6 +484,242 @@ def do2d(
     return _handle_plotting(dataset, do_plot, interrupted())
 
 
+class _AbstractSweep:
+    """
+    Superclass for children sweep classes. Users probably do not want to use
+    this class directly unless they want to define a new child sweep class
+    that is not among sweep classes in this module.
+
+    Args:
+        param: Qcodes _BaseParameter for sweep.
+        start: Sweep start value.
+        stop: Sweep end value.
+        num_points: Number of sweep points.
+        delay: Time in second between two consequtive sweep points. The default
+            is 0.
+    """
+    def __init__(self, param: _BaseParameter, start: float, stop: float,
+                 num_points: int, delay: float = 0):
+        self._param = param
+        self._start = start
+        self._stop = stop
+        self._num_points = num_points
+        self._delay = delay
```

Comment:
Hmn thinking about this perhaps the interface. That way users can define another interface that works differently.

E.g. start, stop and stepsize

Reply:
That was my concern too because I limit people to use this setpoints convension.
I need to get a bit info about intefacing in Python. Then I will change it. Any suggestion to good examples and documents are welcome :)